config/localTemplate.go

```diff
@@ -862,3 +862,37 @@ func (cfg *Local) AdjustConnectionLimits(requiredFDs, maxFDs uint64) bool {
 
 	return true
 }
+
+// StoresCatchpoints returns true if the node is configured to store catchpoints
+func (cfg *Local) StoresCatchpoints() bool {
+	switch cfg.CatchpointTracking {
+	case -1:
+		// No catchpoints.
+	default:
+		fallthrough
+	case 0, 1:
+		if cfg.Archival && (cfg.CatchpointInterval > 0) {
```

Comment:
It took me a few tries to understand this combined switch. I think you got it right.

Maybe constants for each of the cases would help with clarity. Not sure these are the perfect names:
```
const (
trackCatchpointsOff = -1
trackCatchpointsAuto = 0
trackCatchpoints = 1
trackCatchpointsAndStore = 2
)
```

With `CatchpointTracking == 1` and `CatchpointInterval == 1` would the catchpoint be written on every round

Reply:
Went with the following let me know what you think.
```golang
// CatchpointTrackingModeUntracked defines the CatchpointTracking mode that does _not_ track catchpoints
const CatchpointTrackingModeUntracked = -1

// CatchpointTrackingModeAutomatic defines the CatchpointTracking mode that automatically determines catchpoint tracking
// and storage based on the Archival property and CatchpointInterval.
const CatchpointTrackingModeAutomatic = 0

// CatchpointTrackingModeTracked defines the CatchpointTracking mode that tracks catchpoint
// as long as CatchpointInterval > 0
const CatchpointTrackingModeTracked = 1

// CatchpointTrackingModeStored defines the CatchpointTracking mode that tracks and stores catchpoints
// as long as CatchpointInterval > 0
const CatchpointTrackingModeStored = 2
```

Reply:
LGTM